wandb/sdk/wandb_run.py

```diff
@@ -2222,6 +2234,17 @@ def _create_image_job(
         artifact = self.use_artifact(job_artifact)
         return artifact
 
+    def _log_job_artifact_with_image(self, docker_image_name: str) -> Artifact:
+        packages, in_types, out_types = self._make_job_source_reqs()
+        job_artifact = self._create_image_job(
```

Comment:
Add a call to `log_artifact` in here, and remove the calls to `use_artifact` in each of the `_create_*_job` functions and put a use_artifact call in `_log_job`